tests/py/unit/test_analyze_lattice_planes.py

```diff
@@ -0,0 +1,146 @@
+from mat3ra.made.material import Material
+from mat3ra.made.tools.analyze.lattice_planes import CrystalLatticePlanesMaterialAnalyzer
+from mat3ra.made.tools.build.slab.entities import Termination
+from mat3ra.made.tools.convert import from_pymatgen
+from mat3ra.utils import assertion as assertion_utils
+from unit.fixtures.generated.fixtures import HfO2_BULK_MATERIAL, SrTiO3_BULK_MATERIAL
+
+MILLER_INDICES = [(0, 0, 1), (1, 1, 0), (0, 1, 1)]
+SrTiO3_EXPECTED_TERMINATION_FORMULAS = {
+    (0, 0, 1): ["SrO", "TiO2"],
+    (1, 1, 0): ["O2", "SrTiO"],
+    (0, 1, 1): ["O2", "SrTiO"],
+}
+SrTiO3_EXPECTED_TERMINATIONS_WITH_VACUUM = {
+    (0, 0, 1): [
+        Termination.from_string("SrO_Pmmm_2"),
+        Termination.from_string("TiO2_Pmmm_2"),
+    ],
+    (1, 1, 0): [
+        Termination.from_string("O2_Pmmm_2"),
+        Termination.from_string("SrTiO_Pmmm_3"),
+    ],
+    (0, 1, 1): [
+        Termination.from_string("O2_Pmmm_2"),
+        Termination.from_string("SrTiO_Pmmm_3"),
+    ],
+}
+
+SrTiO3_EXPECTED_TERMINATIONS_WITHOUT_VACUUM = {
+    (0, 0, 1): [
+        Termination.from_string("SrO_Pmmm_2"),
+        Termination.from_string("TiO2_Pmmm_2"),
+    ],
+    (1, 1, 0): [
+        None,
+        Termination.from_string("SrTiO_P4/mmm_3"),
+    ],
+    (0, 1, 1): [
+        None,
+        Termination.from_string("SrTiO_P4/mmm_3"),
+    ],
+}
+
+# 3 slabs
+SrTiO3_SLAB_WITH_VACUUM_NAMES = ["Sr3 Ti3 O7", "Sr2 Ti2 O8"]
+# Only one possibility
+SrTiO3_SLAB_WITHOUT_VACUUM_NAMES = [
+    "Sr1 Ti1 O3",
+]
+# shifts for 3 slabs
+SrTiO3_SHIFTS_WITH_VACUUM = [0.25, 0.25]
+# Only one possibility
+SrTiO3_SHIFTS_WITHOUT_VACUUM = [0.0, 0.25]
+
+# shifts for 3 slabs
+HfO2_SHIFTS_WITH_VACUUM = [0.0, 0.25, 0.25]
+HfO2_SHIFTS_WITHOUT_VACUUM = [0.0, 0.0]
+# shifts for 3 slabs
+HfO2_SLAB_WITH_VACUUM_NAMES = [
+    "Hf8 O22",
+    "Hf8 O20",
+    "Hf8 O18",
+    "Hf12 O24",
+    "Hf12 O22",
+    "Hf12 O20",
+    "Hf12 O18",
+    "Hf16 O24",
+]
+HfO2_SLAB_WITHOUT_VACUUM_NAMES = ["Hf4 O8"]
+
+
+def create_analyzer(material, miller_indices):
```

Comment:
I don't see the need for this, let's just use CrystalLatticePlanesMaterialAnalyzer directly. You can import it as `Analyzer` here if needed